moon/rospy/rospy_base.py

```diff
@@ -85,7 +86,12 @@ def callback_clock(self, data):
             self.socket_send(header + to_send)
         self.prev_time = data.clock
 
-    def callback_topic(self, data, topic_name):
+    def callback_topic(self, data, topic_name, rate=1):
+        if topic_name not in self.downsample_topic_count.keys():
```

Comment:
you could use `defaultdict` for `int`, which is automatically initialized to 0

Reply:
You should start using the 'actually enter the suggested change in the code' feature ;)
Good to know and if I knew exactly what to do and the impact of that action, I would do. But things being as they are, I may revisit this next time we are doing something around this code that will need testing instead, at this point I don't want to introduce code change that would invalidate the extensive testing behind the code being committed.